src/TableOperations.jl

```diff
@@ -309,9 +311,33 @@ function joinpartitions(x)
             N = length(schema[].names)
             foreach(i -> push!(joined, ChainedVector([Tables.getcolumn(cols, i)])), 1:N)
         else
-            foreach(i -> append!(joined[i], Tables.getcolumn(cols, i)), 1:N)
+            foreach(1:N) do i
+                prev = joined[i]
+                col = Tables.getcolumn(cols, i)
+                T = eltype(prev)
+                S = eltype(col)
+
+                if !(S <: T) && promote
```

Comment:
With normal `joinpartitions` does the schema's type for a column just indicate the type of the first vector in `ChainedVector`? And that's why (with strict subtypes) `S <: T` is okay but `S >: T` is not?